### ISSUE TITLE
A new voice prompt of "Continue straight ahead" would be useful

### ISSUE THREAD

**Issue by MetalicComputing** [organicmaps/organicmaps#5027]
Hi,

I would like to suggest a new voice prompt being added to Organic Maps. This voice prompt would be to give the instruction to "Continue straight ahead" or similar prompt. I believe this would be useful in situations where there are turnings left and right but also a path ahead of you which is where the Organic Maps app is leading you to go on the screen.
This would be useful in situations where you are listening to voice instructions through headphones to navigate and may not want to pick up your device from a pocket or holder to check the screen. The "Turn Left" and "Turn Right" instructions appear to work. A new voice prompt such as this may also be useful to users who may be visually impaired to navigate to their end destination, as they would be less dependent on the screen of their device for instructions. 

I went out the other day and recreated a route and also did a couple of other routes with the app and have gathered some screenshots of my route on the Organic Maps app and some local photographs of these points for reference if it is helpful. I have marked where the app was leading me on the photos for context. 

My device is a Nokia T10 running Android 12 if this is useful. 

Many thanks.

**Comment by MetalicComputing** [comment]
Case 1:
![Screenshot_20230422-124040](https://user-images.githubusercontent.com/131595637/233842360-c897ac15-6b15-41a9-94c7-6298cfd5ad83.png)

**Comment by biodranik** [comment]
Thanks! The general idea now in OM is the simplicity, by avoiding unnecessary directions. "No directions" mean continue straight. 

From your feelings, where you could be confused without looking on the map? It makes sense to collect some useful corner cases where people can make mistakes without announcing "continue straight". WDYT?

**Comment by MetalicComputing** [comment]
Maybe this is a niche case but could be useful. For example, if you are in cycling mode and have your device in a phone / tablet holder on your handlebars and its a bright sunny day, you might not be able to see what is on the screen at all. This would mean you are more reliant on voice instructions as you cycle along the road or dedicated cycle path. 
Perhaps in the walking mode it is not needed as much and is just reassurance if you are going through an area you are not familiar with and don't want to get your device out of a pocket / holder / backpack every do often.
Maybe if there was this voice instruction included, but you could turn it on and off in the settings. That way people who would like it could use it and those who want to keep that simpler idea of no prompt means just keep going are both happy.

**Comment by ZippyGerbil** [comment]
Agreed, I was quite surprised this was missing. 

I think particularly in busy/urban environments when driving I don't look at the screen at all other than turning at junctions, and reassurance after a turn that I don't need to look at the screen for 5/10 minutes would be a big plus. Otherwise I start to worry I've missed a turn instruction or the phone is acting up.

**Comment by biodranik** [comment]
Wait, we can't announce "continue straight" at every crossing. It is too annoying, did you try that?

I'm still trying to understand when it is important. 

1. If you hear a direction, then you follow it. 
2. If you don't hear a direction, then you continue straight. 

When this behavior should be changed and why?

**Comment by MetalicComputing** [comment]
I think it probably be most useful in my first example. This is the main one I was wanting to point out and which gave me the idea the other week. I went back to screenshot it for the example. 
In this case I was on one distinct street / road and my route was taking me straight ahead, which was the start of a different distinct street / road, of which to get to, I had to cross another road which was a bit more of a main road. (All be it in this case that was a B road). If it were the same road or route continuously which you were following for a few miles, it probably wouldn't be necessary then. But I can also understand ZippyGerbil's point.

**Comment by biodranik** [comment]
Isn't it the same road/way in OpenStreetMap.org data?

**Comment by ZippyGerbil** [comment]
> Wait, we can't announce "continue straight" at every crossing. It is too annoying, did you try that?
> 
> I'm still trying to understand when it is important. 
> 
> 1. If you hear a direction, then you follow it. 
> 2. If you don't hear a direction, then you continue straight. 
> 
> When this behavior should be changed and why?

No absolutely, sorry I didn't mean every crossing.

My experience of other solutions (Google Maps I believe does this), is that when you turn onto a new road, if you don't have another immediate turn instruction, you are given a distance for how long you are on that new road for. E.g. 'Continue on New Road for 1km'.

This sets an expectation for when to expect the next turn instruction, so you can mentally tune back into the radio/surroundings/whatever without being on edge that you might miss something. It also means in a highway environment you know whether you have enough distance to bother getting into the 'fast' lane without looking at the screen.

**Comment by biodranik** [comment]
Agree. That may be useful in specific scenarios like you described (a new road, or a complex crossing). The proper design should:
1. Investigate all use cases for different route types (cars, walking, bicycles).
2. Make a list of explicit rules for the algorithm to generate that "continue straight" direction.
